Sprint 9 solution http api

gennadiyniki opened the pull request:
ФЗ9

gennadiyniki committed:
ФЗ9

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java

VadimZharkov commented on src/httptaskserver/HttpTaskServer.java:
```diff
@@ -0,0 +1,45 @@
+    private final HttpServer httpServer;
+    private final TaskManager taskManager;
+
+    public HttpTaskServer(TaskManager taskManager) throws IOException {
```
throws IOException в сигнатуре метода указывать не нужно, т.к. в методе ты его перехватил и выбросил RuntimeException, которое является unchecked exception, а они в сигнатуре метода не указываются.

VadimZharkov commented on src/httptaskserver/HttpTaskServer.java:
```diff
@@ -0,0 +1,45 @@
+        TaskManager taskManager = Managers.getDefaultManager();
+        HttpTaskServer taskServer = new HttpTaskServer(taskManager);
+        taskServer.start();
+        taskServer.stop();
```
stop(); наверное лишнее, т.к. только что стартанули.

VadimZharkov commented on src/httptaskserver/BaseHttpHandler.java:
```diff
@@ -0,0 +1,105 @@
+        exchange.close();
+    }
+
+    public BaseHttpHandler(TaskManager taskManager) {
```
Конструктор принято размещать после полей до остальных методов

VadimZharkov commented on src/httptaskserver/BaseHttpHandler.java:
```diff
@@ -0,0 +1,105 @@
+import java.time.LocalDateTime;
+
+public class BaseHttpHandler implements HttpHandler {
+    protected TaskManager taskManager;
```
final

VadimZharkov commented on src/httptaskserver/BaseHttpHandler.java:
```diff
@@ -0,0 +1,105 @@
+
+public class BaseHttpHandler implements HttpHandler {
+    protected TaskManager taskManager;
+    protected Gson gson;
```
final

VadimZharkov commented on src/httptaskserver/TasksHandler.java:
```diff
@@ -0,0 +1,94 @@
+            } catch (Exception.NotFoundException e) {
+                sendNotFound(exchange, "Такой Таски нет");
+            }
+        }
```
Нет обработки ситуации пересечения по времени

gennadiyniki replied:
добавил!

VadimZharkov commented on src/httptaskserver/TasksHandler.java:
```diff
@@ -0,0 +1,94 @@
+public class TasksHandler extends BaseHttpHandler {
+    public TasksHandler(TaskManager taskManager) {
+        super(taskManager);
+        //this.taskManager = taskManager;
```
Закомментированный код убираем

VadimZharkov commented on src/httptaskserver/SubtasksHandler.java:
```diff
@@ -0,0 +1,93 @@
+public class SubtasksHandler extends BaseHttpHandler {
+    public SubtasksHandler(TaskManager taskManager) {
+        super(taskManager);
+        //this.taskManager = taskManager;
```
Закомментированный код убираем

VadimZharkov commented on src/httptaskserver/SubtasksHandler.java:
```diff
@@ -0,0 +1,93 @@
+            } catch (Exception.NotFoundException e) {
+                sendNotFound(exchange, "Такой Сабтаски нет");
+            }
+        }
```
Нет обработки ситуации пересечения по времени

VadimZharkov commented on src/manager/Exception.java:
```diff
@@ -0,0 +1,22 @@
+        super(message);
+    }
+
+    public class NotFoundException extends RuntimeException {
```
Лучше разместить класс в отдельном файле, а файлы с исключениями сгруппировать в пакете

gennadiyniki replied:
Убрал, изначально посчитал, что в хранить в одном будет удобнее

VadimZharkov commented on test/httptaskserver/HttpTaskTest.java:
```diff
@@ -0,0 +1,169 @@
+
+
+        // Создаем временный файл для каждого теста
+        tempFile = Files.createTempFile("tasks", ".csv");
```
Эта и след. строка лишние

VadimZharkov reviewed (changes requested):
Направление движения верное, нужно исправить замечания.

gennadiyniki committed:
Исправил указанные замечания.
# Conflicts:
#	src/manager/Managers.java
#	test/manager/FileBackedTaskManagerTest.java
#	test/manager/InMemoryTaskManagerTest.java
